File: 2279_maximum_bags_with_full_capacity_of_rocks/solution.py
# LeetCode 2279 - Maximum Bags With Full Capacity of Rocks
# https://leetcode.com/problems/maximum-bags-with-full-capacity-of-rocks/

from typing import List


class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        need = [c - r for c, r in zip(capacity, rocks)]
        need.sort()
        ans = 0
        for n in need:
            if additionalRocks < n:
                break
            additionalRocks -= n
            ans += 1
        return ans

File: 2279_maximum_bags_with_full_capacity_of_rocks/Solution.scala
// LeetCode 2279 - Maximum Bags With Full Capacity of Rocks
// https://leetcode.com/problems/maximum-bags-with-full-capacity-of-rocks/

object Solution {
  def maximumBags(capacity: Array[Int], rocks: Array[Int], additionalRocks0: Int): Int = {
    val need = new Array[Int](capacity.length)
    var i = 0
    while (i < capacity.length) {
      need(i) = capacity(i) - rocks(i)
      i += 1
    }
    java.util.Arrays.sort(need)
    var additionalRocks = additionalRocks0
    var ans = 0
    for (n <- need) {
      if (additionalRocks < n) return ans
      additionalRocks -= n
      ans += 1
    }
    ans
  }
}
